mne/io/proj.py

```diff
@@ -696,15 +699,36 @@ def make_eeg_average_ref_proj(info, activate=True, verbose=None):
     return eeg_proj
 
 
-def _has_eeg_average_ref_proj(projs):
-    """Determine if a list of projectors has an average EEG ref"""
+def _has_eeg_average_ref_proj(projs, check_active=False):
+    """Determine if a list of projectors has an average EEG ref
+
+    Optionally, set check_active=True to additionally check if the CAR
+    has already been applied.
+    """
     for proj in projs:
         if (proj['desc'] == 'Average EEG reference' or
                 proj['kind'] == FIFF.FIFFV_MNE_PROJ_ITEM_EEG_AVREF):
-            return True
+            if not check_active:
+                return True  # CAR found, don't care if it's active
+            elif proj['active']:  # CAR found, return True only if it's active
+                return True
     return False
 
 
+def _remove_eeg_average_ref_proj(projs):
+    """Remove the EEG average reference projection, if not yet applied
+    """
+    for idp, proj in enumerate(projs):
+        if (proj['desc'] == 'Average EEG reference' or
+                proj['kind'] == FIFF.FIFFV_MNE_PROJ_ITEM_EEG_AVREF):
+            if not proj['active']:
+                projs.pop(idp)
+                logger.info('EEG average reference projection removed.')
```

Comment:
The `_apply_reference` function also removes the CAR if it exists. Now you have refactored to CAR-removal code into a separate function, can you update `_apply_reference` to call this function as well? For consistency, the message should be: "Removing existing average EEG reference projection."